app/Domain/Goalcanvas/Services/Goalcanvas.php

```
@@ -202,5 +223,488 @@ public function pollForUpdatedGoals(): array|false
+            } else {
+                $currentCanvasId = $allCanvas[0]['id'] ?? -1;
+            }
+
+            session([$sessionKey => $currentCanvasId]);
+
+            return $currentCanvasId;
+        }
+
+        private function validateCanvasId(int $canvasId, array $allCanvas): int
+        {
+            foreach ($allCanvas as $canvas) {
+                if ($canvasId == $canvas['id']) {
+                    return $canvasId;
+                }
+            }
+            return -1;
+        }
+
+        // Goal Dashboard Post
+
+        public function handleDashboardPostRequest($params): array
+        {
+            $result = [
+                'redirect' => false,
+                'redirectUrl' => '',
+                'notification' => null,
+            ];
+
+            $currentCanvasId = session("current" . strtoupper(static::CANVAS_NAME) . "Canvas") ?? -1;
+
+            if (isset($params['newCanvas'])) {
+                $result = $this->handleNewCanvas($params, $result);
+            } elseif (isset($params['editCanvas']) && $currentCanvasId > 0) {
+                $result = $this->handleEditCanvas($params, $currentCanvasId, $result);
+            } elseif (isset($params['cloneCanvas']) && $currentCanvasId > 0) {
+                $result = $this->handleCloneCanvas($params, $currentCanvasId, $result);
+            } elseif (isset($params['mergeCanvas']) && $currentCanvasId > 0) {
+                $result = $this->handleMergeCanvas($params, $currentCanvasId, $result);
+            } elseif (isset($params['importCanvas'])) {
+                $result = $this->handleImportCanvas($params, $result);
+            }
+
+            return $result;
+        }
+
+        private function handleNewCanvas($params, $result): array
+        {
+            if (!isset($params['canvastitle']) || empty($params['canvastitle'])) {
+                return $this->setNotification($result, 'notification.please_enter_title', 'error');
```

This should throw a MissingParameterException()

---

`throw new MissingParameterException()`